packages/calypso-e2e/src/lib/blocks/block-components/template-part-block.ts

```diff
@@ -0,0 +1,44 @@
+import { Locator, Page } from 'playwright';
+
+const selectors = {
+	chooseButton: 'button:has-text("Choose")',
+	startBlankButton: 'button:has-text("Start blank")',
+};
+
+/**
+ * Represents a Template Part block in the full site editor.
+ */
+export class TemplatePartBlock {
+	static blockName = 'Template Part';
+	static blockEditorSelector = '[aria-label="Block: Template Part"]';
+
+	private page: Page;
+	block: Locator;
+
+	/**
+	 * Creates an instance of the component.
+	 *
+	 * @param {Page} page Object representing the base page.
+	 * @param {Locator} block Frame-safe locator to the top element of the block.
+	 */
+	constructor( page: Page, block: Locator ) {
+		this.page = page;
+		this.block = block;
```

Comment:
Want to try using the new construction that we learned? haha

Reply:
For sure, I'm all about it! I'll do that in all our new classes!